lessons/micropython/mini-workshop/index.md

```diff
@@ -0,0 +1,319 @@
+
+Dokážeš výsledek dát do cyklu, aby program na zmáčknutí či puštění tlačítka
+reagoval automaticky?
+
+
+## Barvy
+
+Poslední věc připojená k destičce je LED pásek s *několika* světýlkama.
+Každé z nich se dá nastavit na nějakou barvu.
+Zkus si to:
+
+```pycon
+>>> strip[0] = RED
+>>> strip[1] = GREEN
+>>> strip[2] = BLUE
+>>> strip.write()
+```
+
+První tři příkazy připravují jednotlivé barvy, a poslední všechno připravené
+pošle do LED pásku.
+Až budeš experimentovat, nezapomeň na `strip.write()` – bez toho
+se barvy neukážou.
+
+<div style="page-break-after: always;"></div>
+
+## Program
+
+<img src="{{ static('icon_gedit.png')}}" style="float:right;height:5em;">
+
+Teď si přestaneme jen tak hrát, a začneme psát program.
+
+Spusť *editor*: Klikni na *Activities*; potom vyber z levého panelu Gedit
+(ikonka zápisníku s tužkou).
+
+Objeví se bílé okýnko, do kterého napiš:
+
+```python
+strip[0] = WHITE
+strip[1] = OFF
+strip[2] = OFF
+strip.write()
+```
+
+Soubor ulož (*Save*) pod jménem `main.py`.
+
+Potom se vrať k příkazové řádce a ukonči `picocom`:
+stiskni Ctrl+A a pak Ctrl+Q.
+Měl by se objevit řádek končící dolarem.
+
+Za dolar napiš:
+
+```bash
+ampy -p /dev/ttyUSB0 run main.py
+```
+
+... a stiskni Enter. Program se spustí.
+
+Zkus v editoru nastavit jiné barvy a program znovu uložit (Ctrl+S).
+
+Potom v příkazové řádce zmáčkni *šipku nahoru*.
+Tím se vrátíš k předchozímu příkazu (ampy), a můžeš znovu stisknout Enter
+a program spustit.
+
+
+## Blikání
+
+Tady je program, který rozbliká LED pásek.
+Ukazuje větší cyklus, a navíc funkci `sleep`, která MicroPython na chvíli
+zdrží: `sleep(1/4)` čeká čtvrtinu sekundy.
+
+Při přepisování dávej pozor na odsazení – řádky vevnitř v cyklu musí být
+všechny odsazené o čtyři mezery.
+
+```python
+strip[0] = OFF
+strip[1] = OFF
+strip[2] = OFF
+
+while True:
+    strip[0] = RED
+    strip[1] = OFF
+    strip.write()
+    sleep(1/4)
+
+    strip[0] = OFF
+    strip[1] = RED
+    strip.write()
+    sleep(1/4)
+```
+
+## Semafor
+
+<img src="{{ static('semafor.gif')}}" style="float:right;height:5em;">
+
+Dokážeš naprogramovat semafor?
+
+Se závorou?
+
+{# <div style="page-break-after: always;"></div> #}
+
+## Cyklus s podmínkou
+
+Cykly nemusí být jen nekonečné.
+Tady je cyklus, který čeká dokud není stisknuté tlačítko.
+
+Potom se rozsvítí světlo – všimni si, že příkaz `led.value(1)` už není
+odsazený. Není součást cyklu.
+
+```python
+led.value(0)
+
+while btn.value():
+    sleep(1/100)
+
+led.value(1)
+```
+
+
+## Alarm
+
+Polož na tlačítko nějaký těžší předmět (třeba myš) tak, aby bylo zmáčknuté.
+Když někdo předmět vezme, tlačítko přestane být zmáčknuté.
+
+Napiš program, který na to bude čekat, a jakmile někdo myš ukradne,
+LED pásek začne zuřivě blikat.
+
+
+## Konec?
+
+Tím dnešní krátká exkurze do světa MicroPythonu končí.
+Jestli tě programování zaujalo a chceš v něm pokračovat, dej nám vědět!
+
+Nebo pak z klidu domova napiš Petrovi na pviktori@redhat.com.
```

Comment:
Ne, materiály dostali účastníci vytištěné. Ale nebude mi vadit když se to v budoucnu linkovat začne.